compaction/compaction_manager.cc

```diff
@@ -695,6 +697,10 @@ future<> compaction_manager::update_throughput(uint32_t value_mbs) {
 void compaction_manager::register_metrics() {
     namespace sm = seastar::metrics;
 
+    sm::label mode_label("mode");
+    const auto segregate_mode_label_instance = mode_label("segregate");
+    const auto validate_mode_label_instance = mode_label("validate");
```

Comment:
let's separate the metrics part into a patch of its own.
there's already a lot going on in this patch without that.